wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireInput.kt

```diff
@@ -23,24 +23,14 @@ import java.io.RandomAccessFile
 import org.gradle.api.internal.file.FileOperations
 
 internal val List<ProtoRootSet>.inputLocations: List<InputLocation>
-  get() = flatMap { rootSet ->
-    rootSet.roots.map { file ->
-      rootSet.inputLocation(file)
-    }
-  }
+  get() = map { rootSet -> rootSet.inputLocation() }
```

Comment:
This avoids creating an InputLocation per-file, but intead creates an InputLocation per-ProtoRootSet, with all the files in the ProtoRootSet treated as a "classpath"